src/Mod/Sketcher/App/Sketch.cpp

```diff
@@ -2868,6 +2868,11 @@ int Sketch::addTangentConstraint(int geoId1, int geoId2)
             GCSsys.addConstraintTangent(l, a, tag);
             return ConstraintsCounter;
         }
+        else if (Geoms[geoId2].type == BSpline) {
+            Base::Console().Error("Direct tangency constraint between line and B-spline is not "
+                                  "supported. Use tangent-via-point instead.");

```

Comment:
I think this is more of a "pop-up window" sort of error, not just a message in the report view. Maybe the DWG can weigh in? @obelisk79?

Reply:
What's DWG?

Reply:
@AjinkyaDahale Design Working Group. We try to provide advice to developers for UI/UX related items using some established guidelines. It's a relatively new thing for FreeCAD.

@chennes I agree, adding this to the non-obtrusive alerts would be better. A pop-up window would likely be too disruptive. I'll bring this up with the group on Discord.

Reply:
I'm not sure some disruption isn't a _good_ idea here: generally speaking, you're only going to do it wrong once... after that single disruption, you've (hopefully!) learned how to use curve-to-spline tangency -- e.g. you have to tell the system at what point you want it to be tangent, and are unlikely to use it incorrectly again. It could even be one of those dialogs with the "Don't show this again" checkbox.

Reply:
Yeah, I concur with a popup. The "Don't show this again" checkbox usually shows for non-critical popups. Like redundant constraint removal etc.

Reply:
I think there doesn't need to be a pop-up. Also, someone can ignore the text, tick "don't ask again", and then forever get confused. A nonobstrusive but revisitable error is better. Ideally it should ne accompanied with the constraint being deleted, but I'm still not comfortable with that part of the code.

Reply:
I agree about not including a checkbox. Regarding unobtrusive vs pop-up. I suppose the criticality, and gracefulness of the error state/handling should dictate which is more appropriate. If the constraint is removed and nothing happens negatively to the sketch other than a notification, avoiding the popup is ok. But a popup dialog should appear very infrequently if implemented as such. Both approaches to alerting the user of a problem seem reasonably acceptable regarding the above criteria.

Reply:
Sending errors to report view also triggers the non intrusive notifications. The notifications system is a very good idea but it really needs polishing. It would be great if the DWG could have a close look at it, identify the problems and write up some guidelines that devs could follow to make them better.

Sorry for the off topic comment!